grid-demo/src/app/core/column-ordering/column-ordering.component.ts

```diff
@@ -1,7 +1,74 @@
 import { Component } from '@angular/core';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { debounceTime, first, map, publishReplay, refCount, shareReplay, switchMap } from 'rxjs/operators';
+import '@cds/core/grid/register.js';
+import '@cds/core/checkbox/register.js';
+import '@cds/core/dropdown/register.js';
+
+import { columns, Vm, generateVms } from '../../data/vm.generator';
 
 @Component({
   templateUrl: './column-ordering.component.html',
   styleUrls: ['./column-ordering.component.scss'],
 })
-export class CoreColumnOrderingComponent {}
+export class CoreColumnOrderingComponent {
+  readonly pageIndex = new BehaviorSubject<number>(0);
+  readonly pageSize = new BehaviorSubject<number>(10);
+  readonly vms: Observable<Vm[]>;
+  readonly pageCount: Observable<number>;
+  readonly totalResults = 1000;
+  readonly columns = new BehaviorSubject<typeof columns>(columns);
+
+  readonly ariaLiveMessage = new BehaviorSubject<string>('');
+
+  constructor() {
+    const allVms = this.getAllVms().pipe(shareReplay(1));
+    this.vms = this.getVms(allVms).pipe(shareReplay(1));
+    this.pageCount = this.getPageCount().pipe(shareReplay(1));
+  }
+
+  pageChange(event: any) {
+    this.pageIndex.next(event.detail);
+  }
+
+  pageSizeChange(event: any) {
+    this.pageSize.next(event.detail);
+    this.pageIndex.next(0);
+  }
+
+  sortColumns(e: any) {
+    console.log(e);
+    if (e.detail.type === 'reordered') {
+      const value = [...this.columns.value];
+
+      const fromIndex = value.findIndex(i => `${i.field}` === e.detail.from.getAttribute('cds-field'));
```

Comment:
This looks very similar to the logic I had in my reorder columns component. I've since realized that this is not the desired behavior. We don't want to just swap the two columns, we actually want to move one column and "shift" the other columns over. So the logic is more complicated than this. I am now using the `moveItemInArray` function in `@angular/cdk/drag-drop` to achieve this.

Reply:
Oh yeah I totally copied your original implementation here. I grabbed Cory's helper function from the Core repo, but if you're already using `moveItemInArray` that's better